### PR TITLE
Add CPU feature detection for macOS/aarch64.

### PR DESCRIPTION
**Pull request by makotokato**

### CODE REVIEW

**Review by lu-zero — APPROVED** [review]
Looks fine and seems correct on m2

File: crates/std_detect/src/detect/os/macos/aarch64.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+//! Run-time feature detection for aarch64 on macOS.
+
+use crate::detect::{cache, Feature};
+
+#[inline]
+fn _sysctlbyname(name: &str) -> bool {
+    use libc;
+
+    let mut enabled: i32 = 0;
+    let mut enabled_len: usize = 4;
```
Comment (Amanieu) [inline crates/std_detect/src/detect/os/macos/aarch64.rs:10]: Are you sure this is the correct size? I couldn't find anything in Apple's documentation that indicates what size to use.

File: crates/std_detect/src/detect/os/macos/aarch64.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            name.as_ptr() as *const i8,
+            enabled_ptr,
+            &mut enabled_len,
+            core::ptr::null_mut(),
+            0,
+        )
+    };
+
+    match ret {
+        0 => enabled != 0,
```
Comment (Amanieu) [inline crates/std_detect/src/detect/os/macos/aarch64.rs:24]: Should the returned size be checked here as well?

File: crates/std_detect/src/detect/os/macos/aarch64.rs (hunk 1 above)
Comment (makotokato) [reply crates/std_detect/src/detect/os/macos/aarch64.rs:10]: Other proeject such as [Google cpufeature](https://github.com/google/cpu_features/blob/main/src/impl_aarch64_macos_or_iphone.c) and [Apple WebKit](https://searchfox.org/wubkat/source/Source/JavaScriptCore/assembler/CPU.cpp#127) uses `int`.

File: crates/std_detect/src/detect/os/macos/aarch64.rs (hunk 2 above)
Comment (makotokato) [reply crates/std_detect/src/detect/os/macos/aarch64.rs:24]: It is unnecessary to check size as long as I look Apple's WebKit code.

**Pull request merged**